pkg/util/modules/modules_test.go

```diff
@@ -39,13 +46,24 @@ func TestDependencies(t *testing.T) {
 	require.Len(t, invDeps, 1)
 	assert.Equal(t, invDeps[0], "serviceB")
 
-	svcs, err := mm.InitModuleServices("serviceC")
-	assert.NotNil(t, svcs)
+	// Test unknown module
+	svc, err := mm.InitModuleServices("service_unknown")
+	assert.Error(t, err, fmt.Errorf("unrecognised module name: service_unknown"))
+	assert.Empty(t, svc)
+
+	// Test init failure
+	svc, err = mm.InitModuleServices("serviceD")
+	assert.Error(t, err)
+	assert.Empty(t, svc)
+
+	svc, err = mm.InitModuleServices("serviceC")
 	assert.NoError(t, err)
+	assert.Equal(t, 3, len(svc))
 
-	svcs, err = mm.InitModuleServices("service_unknown")
-	assert.Nil(t, svcs)
-	assert.Error(t, err, fmt.Errorf("unrecognised module name: service_unknown"))
+	// Test loading of the module second time (should be noop)
```

Comment:
It should not be noop -- it should in fact create new services. Can we extend the test to verify that services from first and second invocation of `mm.InitModuleServices("serviceC")` are different?

Reply:
Yeah, the noop part is left from the previous stateful ModuleManager. Fixed